operator/v1/types_ingress.go

```diff
@@ -282,6 +287,315 @@ var (
+	//
+	// and looks like this (yaml):
+	//
+	//   cipers:
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.2
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile `json:"modern,omitempty"`
+	// custom is a user-defined security profile. An example custom profile
+	// looks like this:
+	//
+	// cipers: >-
+	//   ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:
+	//   ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256
+	// securityProtocol:
+	//   minimumVersion: TLSv1.1
+	//   maximumVersion: TLSv1.2
+	// dhParamSize: 1024
+	//
+	// +optional
+	// +nullable
+	Custom *CustomSecurityProfile `json:"custom,omitempty"`
+}
+
+type OldSecurityProfile struct{}
+type IntermediateSecurityProfile struct{}
+type ModernSecurityProfile struct{}
+
+type CustomSecurityProfile struct {
+	// securityProfile defines the schema for a custom security profile.
+	//
+	// Any unspecified fields will default to the "Intermediate" compatibility configuration
+	// as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	SecurityProfile `json:"securityProfile"` // inline
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// Old is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// Intermediate is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// Modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+	// Custom is a security profile that allows for user-defined parameters.
+	SecurityProfileCustomType SecurityProfileType = "Custom"
+)
+
+// SecurityProfile defines the schema for a security profile.
+type SecurityProfile struct {
+	// ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Preface a cipher
+	// with a "!" to disable a specific cipher from being negotiated. For example,
+	// to use DES-CBC3-SHA but not EDH-DSS-DES-CBC3-SHA (yaml):
+	//
+	//   cipers:
+	//     - DES-CBC3-SHA
+	//     - !EDH-DSS-DES-CBC3-SHA
+	//
+	// If unset, the "Intermediate" Ciphersuites are used from here:
```

Comment:
I'm not sure this statement is true. The defaulting is contextualized with the type of profile on the SecurityProfileSpec. The value seems to be just as-is.

Reply:
Good catch, yes this type of documentation should be within the profile type. I'll update.